src/utils.jl

```diff
@@ -0,0 +1,98 @@
+using OrderedCollections, SparseArrays
+
+"""
+$(TYPEDSIGNATURES)
+
+Return the differential equations and other non-differential equations.
+
+For differential equations, this function assume the derivative is the only single term at
+the left hand side, which is typically the result of `ModelingToolkit.structural_simplify`.
+
+Equations from subsystems are not included.
```

Comment:
why is that? I would assume the lowered form would be better to use here, `ModelingToolkit.equations(sys)`?

Reply:
`ModelingToolkit.equations` is internally called by `ModelingToolkit.tearing_substitution` in the first step of `deim`.

This helper function `get_deqs` splits differential equations and algebraic equations. I think it doesn't need to check `ODESystem.substitutions` and `ODESystem.systems` once more. Is this ok?

Reply:
I see, so okay tearing will flatten so therefore it's fine.